### ISSUE TITLE
Running Mailcow Dockerized on Docker for Windows?

### ISSUE THREAD

**Issue by jaimbo** [mailcow/mailcow-dockerized#326]
Hi, I currently have the standalone version of Mailcow running in a VM on a Windows box, but seeing as how it hasn't had any updates for a while and you recommend moving to the Docker version. Is it possible to run the Dockerized version inside Docker for Windows?

**Comment by mkuron** [comment]
Docker for Windows in the end just runs a Linux kernel in a HyperV VM, so I don't think it would be much different than on Linux. Mailcow is fairly standard in terms of how it interacts with the Docker platform; the only exception is the ipv6nat container (and the fail2ban container in the dev branch) which directly sets iptables rules on the host -- but you can disable those for a start.

If Mailcow doesn't work on Windows for whatever reason, you could also install Docker inside a Linux VM.

**Comment by jaimbo** [comment]
Perfect I'll give it a bash after work and I'll report back if there are any issues.

Do you think there will be any problems in performance? I'm sure I've seen the Docker Devs state in a few places that Docker for Windows is only meant for development and testing.

**Comment by andryyy** [comment]
I have no idea. :-) I think you may also want to install the "Windows Subsystem for Linux" => https://msdn.microsoft.com/de-de/commandline/wsl/install_guide

**Comment by Braintelligence** [comment]
I'm pretty sure docker for windows existed before the subsystem for Linux was released. Are you sure it always runs on a Linux kernel?

**Comment by jaimbo** [comment]
Seems to be getting stuck on:

Creating mailcowdockerized_clamd-mailcow_1 ... done

**Comment by mkuron** [comment]
The Windows Subsystem is not needed. It provides an API translation layer so that Linux binaries can be executed on top of the Windows kernel. I'm pretty sure that translation wouldn't support the necessary kernel APIs for containers.

Docker on Windows runs a lightweight Linux VM in the background if you're running Linux containers (there is a diagram somewhere in the documentation that shows that). Only if you're running Windows containers you wouldn't need that.

@jaimbo, can you provide any logs? I don't think anyone has ever tried that with Mailcow.

**Comment by jaimbo** [comment]
@mkuron What kind of logs were you after?

I managed to get Mailcow up and running, even managed to get my databases backed up and transferred over from the old install. The problem above was caused by Docker for Windows not pulling the data from the mailcow.conf. So I instead declared all the vars in a .bat and then ran the commands and that sorted it.

The only thing I can't get working now is the actual emails themselves transferred over. I have the VMail folder from the old install backed up. just can't for the life of me get it into dockerized.

**Comment by mkuron** [comment]
The most probable explanation for your variables issue is that .env is a symlink to mailcow.conf. While NTFS supports symlinks, Windows requires elevated privileges to create them, so Git cannot have created it when you cloned Mailcow-Dockerized. For better Windows compatibility, we'd need to get rid of the symlink, though I'm not sure that's possible.

**Comment by mkuron** [comment]
Apparently you can get proper symlink support in recent Windows 10 versions by enabling Developer Mode: https://blogs.windows.com/buildingapps/2016/12/02/symlinks-windows-10/ . Not sure if Git supports it yet, but in the long run the problem will go away.

I can't help you with the vmail issue. Perhaps you can check the mailcow => mailcow-dockerized migration issues or give some additional information on what's going wrong.

**Comment by andryyy** [comment]
I was thinking of the generate_config script.

**Comment by TBK** [comment]
I have solved the env issue by making a ps script #330 

I spin up mailcow with powershell like this:
```
git clone https://github.com/mailcow/mailcow-dockerized
cd mailcow-dockerized
.\generate_config.sh
.\env.ps1
docker-compose up -d
```
